### ISSUE TITLE
varnishtest passes for vtc assert

### ISSUE THREAD

**Issue by nigoroll** [varnishcache/varnish-cache#2165]
seen with printf debug in cache_esi_pase enabled

```
diff --git a/bin/varnishd/cache/cache_esi_parse.c b/bin/varnishd/cache/cache_esi_parse.c
index d2b2306..39ea6bd 100644
--- a/bin/varnishd/cache/cache_esi_parse.c
+++ b/bin/varnishd/cache/cache_esi_parse.c
@@ -37,8 +37,9 @@
 #include "vend.h"
 #include "vgz.h"
 
-//#define Debug(fmt, ...) printf(fmt, __VA_ARGS__)
-#define Debug(fmt, ...) /**/
+#include <stdio.h>
+#define Debug(fmt, ...) printf(fmt, __VA_ARGS__)
+//#define Debug(fmt, ...) /**/
 
 struct vep_state;
```
 
->

```
######## tests/e00019.vtc ########
Assert error in vtc_log_emit(), vtc_log.c line 104:
  Condition(vtclog_left > l) not true.
...
#     top  TEST tests/e00019.vtc passed (0.564)
```

the workaround is to raise `vtc_bufsiz`, but varnishtest should not pass for failed assertions

**Comment by bsdphk** [comment]
This was fixed in 85b8250edecf74f35d9bcc34dfa19dd98c141595

**Comment by hermunn** [comment]
Backport review: This seems to be a regression, and the bug is, as far as I can see, not in 4.1
